### PR TITLE
Sanity check filtering/sorting options in list requests.

### PR DESCRIPTION
**Pull request by neuromage**
If sort by or filtering criteria is specified in conjunction with a next
page token, ensure they match up, otherwise return an error.

Also, change the errors to be InvalidInputErrors instead of standard Go
string errors to be consistent with the rest of apiserver.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/625)

### CODE REVIEW

File: backend/src/apiserver/server/experiment_server.go [hunk 1]
```diff
@@ -39,19 +38,7 @@ func (s *ExperimentServer) GetExperiment(ctx context.Context, request *api.GetEx
...
-	} else {
-		var f *api.Filter
-		f, err = parseAPIFilter(request.Filter)
-		if err != nil {
-			return nil, err
-		}
-
-		opts, err = list.NewOptions(&model.Experiment{}, int(request.PageSize), request.SortBy, f)
-	}
+	opts, err := validatedListOptions(&model.Experiment{}, request.PageToken, int(request.PageSize), request.SortBy, request.Filter)
```
Comment (IronPan) [inline backend/src/apiserver/server/experiment_server.go]: nit consider making &model.Experiment{} a constant so it doesn't instantiate a new object for each API call.

Comment (neuromage) [reply backend/src/apiserver/server/experiment_server.go]: Hmm, I think any performance issue here is negligible, and the benefits of having this right here for readability (vs another indirection for the reader) outweighs it anyway. It's really cheap to create it and the performance cost of this call is elsewhere, so I'd rather leave it as is. What do you think?

Comment (IronPan) [reply backend/src/apiserver/server/experiment_server.go]: sgtm

**Comment by neuromage** [comment]
Fixed merge conflicts. PTAL.

**Pull request merged**